webhooks/appmesh/virtualnode_validator.go

```diff
@@ -62,7 +68,8 @@ func (v *virtualNodeValidator) enforceFieldsImmutability(vn *appmesh.VirtualNode
 	if !reflect.DeepEqual(vn.Spec.MeshRef, oldVN.Spec.MeshRef) {
 		changedImmutableFields = append(changedImmutableFields, "spec.meshRef")
 	}
-	if oldVN.Spec.ServiceDiscovery.AWSCloudMap != nil && !reflect.DeepEqual(vn.Spec.ServiceDiscovery.AWSCloudMap, oldVN.Spec.ServiceDiscovery.AWSCloudMap) {
+	if oldVN.Spec.ServiceDiscovery != nil && oldVN.Spec.ServiceDiscovery.AWSCloudMap != nil &&
```

Comment:
Let's please extend the virtual node integration to test for cases where service discovery is not specified